tests/common/gcp_type/firewall_rule_test.py

```diff
@@ -173,11 +222,13 @@ def test_firewall_rule_lt(self, rule_1_dict, rule_2_dict, expected):
                 'firewall_rule_source_ranges': json.dumps(['0.0.0.0/0']),
                 'firewall_rule_direction': 'ingress',
                 'firewall_rule_network': 'n1',
+                'firewall_rule_allowed': json.dumps('*'),
```

Comment:
These should be json.dumps(['*']). The reason the code still works is it treats the string '*' as a list and checks each character against the MATCH_ANY value. It works by accident since MATCH_ANY is a single character, but the code should fail.

You probably should add an assert that firewall_rules is a list in FirewallAction init.